File: network/src/test/kotlin/org/bitlap/network/test/HandleTests.kt
package org.bitlap.network.test

import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.shouldBe
import org.bitlap.network.core.HandleIdentifier
import org.bitlap.network.core.OperationType
import org.bitlap.network.core.SessionHandle
import org.bitlap.network.core.operation.OperationHandle

/**
 *
 * @author 梦境迷离
 * @since 2021/9/6
 * @version 1.0
 */
class HandleTests : StringSpec({

    "test operationHandle" {
        val handleId = HandleIdentifier()
        val op = OperationHandle(OperationType.EXECUTE_STATEMENT, hasResultSet = true, handleId = handleId)

        (op.toBOperationHandle().operationId == handleId.toBHandleIdentifier()) shouldBe true

        (op.toBOperationHandle().operationId.guid == handleId.toBHandleIdentifier().guid) shouldBe true

        (op.toBOperationHandle().operationId.secret == handleId.toBHandleIdentifier().secret) shouldBe true

        (op.toBOperationHandle().operationTypeValue == OperationType.EXECUTE_STATEMENT.toBOperationType().number) shouldBe true

        (OperationHandle(op.toBOperationHandle()) == op) shouldBe true
    }

    "test sessionHandle" {
        val handleId = HandleIdentifier()
        val op = SessionHandle(handleId)

        (op.toBSessionHandle().sessionId == handleId.toBHandleIdentifier()) shouldBe true

        (op.toBSessionHandle().sessionId.guid == handleId.toBHandleIdentifier().guid) shouldBe true

        (op.toBSessionHandle().sessionId.secret == handleId.toBHandleIdentifier().secret) shouldBe true

        (SessionHandle(op.toBSessionHandle()) == op) shouldBe true
    }
})

File: network/src/test/scala/org/bitlap/network/test/HandleTests.scala
/* Copyright (c) 2022 bitlap.org */
package org.bitlap.network.test

import org.bitlap.network.types.handles.{ HandleIdentifier, SessionHandle }
import org.bitlap.network.types.{ OperationType, handles }
import org.scalatest.flatspec.AnyFlatSpec
import org.scalatest.matchers.should.Matchers

/**
 * @author 梦境迷离
 * @since 2021/11/21
 * @version 1.0
 */
class HandleTests extends AnyFlatSpec with Matchers {

  "test operationHandle" should "ok" in {
    val handleId = new handles.HandleIdentifier()
    val op = new handles.OperationHandle(
      OperationType.EXECUTE_STATEMENT,
      hasResultSet = true,
      handleId = handleId
    )

    (op.toBOperationHandle().getOperationId == handleId
      .toBHandleIdentifier()) shouldBe true

    (op.toBOperationHandle().getOperationId.getGuid == handleId
      .toBHandleIdentifier()
      .getGuid) shouldBe true

    (op.toBOperationHandle().getOperationId.getSecret == handleId
      .toBHandleIdentifier()
      .getSecret) shouldBe true

    op.toBOperationHandle().getOperationTypeValue == OperationType.EXECUTE_STATEMENT.id shouldBe true

    val h1 = new handles.OperationHandle(op.toBOperationHandle())
    h1 == op shouldBe true
  }

  "test sessionHandle" should "ok" in {
    val handleId = new HandleIdentifier()
    val op = new SessionHandle(handleId)

    (op.toBSessionHandle().getSessionId == handleId
      .toBHandleIdentifier()) shouldBe true

    (op.toBSessionHandle().getSessionId.getGuid == handleId
      .toBHandleIdentifier()
      .getGuid) shouldBe true

    (op.toBSessionHandle().getSessionId.getSecret == handleId
      .toBHandleIdentifier()
      .getSecret) shouldBe true

    new SessionHandle(op.toBSessionHandle()) == op shouldBe true
  }
}
